version: "3.7"

networks:
  net:
    name: fabric_network

services:

  explorerdb:
    image: hyperledger/explorer-db:latest
    container_name: explorerdb
    hostname: explorerdb
    environment:
      - DATABASE_DATABASE=fabricexplorer
      - DATABASE_USERNAME=hppoc
      - DATABASE_PASSWORD=password
    healthcheck:
      test: "pg_isready -h localhost -p 5432 -q -U postgres"
      interval: 30s
      timeout: 10s
      retries: 5
    ports:
      - 4432:5432
    volumes:
      - ../docker_data/explorer_ishare:/var/lib/postgresql/data
    networks:
      - net

  explorer:
    image: hyperledger/explorer:latest
    container_name: explorer
    hostname: explorer
    environment:
      - DATABASE_HOST=explorerdb
      - DATABASE_DATABASE=fabricexplorer
      - DATABASE_USERNAME=hppoc
      - DATABASE_PASSWD=password
      - LOG_LEVEL_APP=debug
      - LOG_LEVEL_DB=debug
      - LOG_LEVEL_CONSOLE=info
      - LOG_CONSOLE_STDOUT=true
      - DISCOVERY_AS_LOCALHOST=false
    volumes:
      - ./explorer-config.json:/opt/explorer/app/platform/fabric/explorer-config.json
      - ./:/opt/explorer/app/platform/fabric/connection-profile
#      - /fabric-path/fabric-samples/test-network/organizations:/tmp/crypto
      - ../channel/crypto-config/:/tmp/crypto
      - ../docker_data/walletstore_ishare:/opt/explorer/wallet
    ports:
      - 8081:8080
    depends_on:
      explorerdb:
        condition: service_healthy
    networks:
      - net    